src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -1018,6 +1522,16 @@ public DateTimeKind Kind
         //
         public int Millisecond => (int)((UTicks / TicksPerMillisecond) % 1000);
 
+        /// <summary>
+        /// The microseconds component, expressed as a value between 0 and 999.
+        /// </summary>
+        public int Microsecond => (int)((UTicks / TicksPerMicrosecond) % 1000);
```

Comment:
As described in https://github.com/deeprobin/runtime/commit/5de0bfb710285fb54e62fd0cbb381f5ca6901d8a#r70621974 this calculation is wrong. This causes test failures.

@tarekgh Some suggestions to fix?

```
      System.Tests.DateTimeOffsetTests.MaxValue [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   1704263580
        Stack Trace:
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(1151,0): at System.Tests.DateTimeOffsetTests.VerifyDateTimeOffset(DateTimeOffset dateTimeOffset, Int32 year, Int32 month, Int32 day, Int32 hour, Int32 minute, Int32 second, Int32 millisecond, Int32 microsecond, Nullable`1 offset)
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(17,0): at System.Tests.DateTimeOffsetTests.MaxValue()
      System.Tests.DateTimeOffsetTests.Ctor_Int_Int_Int_Int_Int_Int_Int_Int_TimeSpan [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   -1993306240
        Stack Trace:
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(1151,0): at System.Tests.DateTimeOffsetTests.VerifyDateTimeOffset(DateTimeOffset dateTimeOffset, Int32 year, Int32 month, Int32 day, Int32 hour, Int32 minute, Int32 second, Int32 millisecond, Int32 microsecond, Nullable`1 offset)
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(161,0): at System.Tests.DateTimeOffsetTests.Ctor_Int_Int_Int_Int_Int_Int_Int_Int_TimeSpan()
      System.Tests.DateTimeOffsetTests.Ctor_DateTime [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   -524812288
        Stack Trace:
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(1151,0): at System.Tests.DateTimeOffsetTests.VerifyDateTimeOffset(DateTimeOffset dateTimeOffset, Int32 year, Int32 month, Int32 day, Int32 hour, Int32 minute, Int32 second, Int32 millisecond, Int32 microsecond, Nullable`1 offset)
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(37,0): at System.Tests.DateTimeOffsetTests.Ctor_DateTime()
      System.Tests.DateTimeOffsetTests.Ctor_DateTime_TimeSpan [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   1704263580
        Stack Trace:
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(1151,0): at System.Tests.DateTimeOffsetTests.VerifyDateTimeOffset(DateTimeOffset dateTimeOffset, Int32 year, Int32 month, Int32 day, Int32 hour, Int32 minute, Int32 second, Int32 millisecond, Int32 microsecond, Nullable`1 offset)
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(84,0): at System.Tests.DateTimeOffsetTests.Ctor_DateTime_TimeSpan()
      System.Tests.DateTimeOffsetTests.ToLocalTime_MinValue [SKIP]
        Condition(s) not met: "IsMinValueNegativeLocalOffset"
      System.Tests.DateTimeOffsetTests.Ctor_Long_TimeSpan [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   -1115569216
        Stack Trace:
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(1151,0): at System.Tests.DateTimeOffsetTests.VerifyDateTimeOffset(DateTimeOffset dateTimeOffset, Int32 year, Int32 month, Int32 day, Int32 hour, Int32 minute, Int32 second, Int32 millisecond, Int32 microsecond, Nullable`1 offset)
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(137,0): at System.Tests.DateTimeOffsetTests.Ctor_Long_TimeSpan()
      System.Tests.DateTimeOffsetTests.AddMicroseconds(dateTimeOffset: 1986-08-15T10:20:05.0700700+00:00, microseconds: -10, expected: 1986-08-15T10:20:05.0700600+00:00) [FAIL]
        System.ArgumentOutOfRangeException : Value to add was out of range. (Parameter 'value')
        Stack Trace:
             at System.DateTime.<AddMicroseconds>g__ThrowOutOfRange|87_0()
             at System.DateTimeOffset.AddMicroseconds(Double microseconds)
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(528,0): at System.Tests.DateTimeOffsetTests.AddMicroseconds(DateTimeOffset dateTimeOffset, Double microseconds, DateTimeOffset expected)
      System.Tests.DateTimeOffsetTests.UnixEpoch [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   -1025114112
        Stack Trace:
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(1151,0): at System.Tests.DateTimeOffsetTests.VerifyDateTimeOffset(DateTimeOffset dateTimeOffset, Int32 year, Int32 month, Int32 day, Int32 hour, Int32 minute, Int32 second, Int32 millisecond, Int32 microsecond, Nullable`1 offset)
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(1418,0): at System.Tests.DateTimeOffsetTests.UnixEpoch()
      System.Tests.DateTimeOffsetTests.Ctor_Int_Int_Int_Int_Int_Int_Int_TimeSpan [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   -1993706240
        Stack Trace:
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(1151,0): at System.Tests.DateTimeOffsetTests.VerifyDateTimeOffset(DateTimeOffset dateTimeOffset, Int32 year, Int32 month, Int32 day, Int32 hour, Int32 minute, Int32 second, Int32 millisecond, Int32 microsecond, Nullable`1 offset)
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(168,0): at System.Tests.DateTimeOffsetTests.Ctor_Int_Int_Int_Int_Int_Int_Int_TimeSpan()
      System.Tests.DateTimeOffsetTests.Ctor_Int_Int_Int_Int_Int_Int_TimeSpan [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   1801261056
        Stack Trace:
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(1151,0): at System.Tests.DateTimeOffsetTests.VerifyDateTimeOffset(DateTimeOffset dateTimeOffset, Int32 year, Int32 month, Int32 day, Int32 hour, Int32 minute, Int32 second, Int32 millisecond, Int32 microsecond, Nullable`1 offset)
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(234,0): at System.Tests.DateTimeOffsetTests.Ctor_Int_Int_Int_Int_Int_Int_TimeSpan()
      System.Tests.DateTimeOffsetTests.ImplicitCast_DateTime [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   -524812288
        Stack Trace:
          E:\external\dotnet\runtime\src\libraries\System.Runtime\tests\System\DateTimeOffsetTests.cs(1151,0): at System.Tests.DateTimeOffsetTests.VerifyDateTimeOffset(DateTimeOffset dateTimeOffset, Int32 year, Int32 month, Int32 day, Int32 hour, Int32 minute, Int32 second, Int32 millisecond, Int32 microsecond, Nullable`1 offset)
```

Reply:
@deeprobin 

When checking the NanoSeconds in the DateTimeOffset.MaxValue, you are checking it to be `0` while this should be `900`. 

I added a comment on DateTime.Nanoseconds for how you can correctly compute it.